### ISSUE TITLE
Import link website & preview

### ISSUE THREAD

**Issue by Aurnytoraink** [flxzt/rnote#913]
**Is your feature request related to a problem? Please describe.**  
Currently, when we paste a link, it's consider as plain text. However, I want to be able to add links pointing to websites in my notes.

**Describe the solution you'd like**  
The link should appear as a block with the URL or the preview, if available. User should be able to visit the link when they click on the note (or an option should appear)

**Describe alternatives you've considered**  
For the link preview, you should take a look at the [Share Preview app](https://flathub.org/fr/apps/com.rafaelmardojai.SharePreview)

**Additional context**  
User may link a URL to a picture, media, icon

**Comment by flxzt** [comment]
Oh adding a webview that previews the link will be unlikely. We could add a link content type that opens the browser when clicked.

**Comment by Aurnytoraink** [comment]
> Oh adding a webview that previews the link will be unlikely.

I don't think you need a webview for the preview. In fact, you can just make a request to the website and scrap the data like [Share Preview app](https://flathub.org/fr/apps/com.rafaelmardojai.SharePreview) does and then display the result in a custom widget, with an image cover, the title and the URL.
